components/motor/gpio/mathutils.go

```diff
@@ -34,12 +32,7 @@ func goForMath(maxRPM, rpm, revolutions float64) (float64, time.Duration) {
 		rpm = -1 * maxRPM
 	}
 
-	if revolutions == 0 {
-		powerPct := rpm / maxRPM
-		return powerPct, 0
-	}
-
-	dir := rpm * revolutions / math.Abs(revolutions*rpm)
+	dir := sign(rpm * revolutions)
```

Comment:
lets return early `if dir == 0`, so we cover both 0 revolutions and 0 rpm cases

Reply:
I think we just need to check this logic everywhere and return early with an error, it's the simplest form of the rpm or revolution being zero check.

Reply:
agreed, now the speed and revs are checked everywhere before this